src/Orleans.Core/Core/ClusterClient.cs

```diff
@@ -20,6 +22,7 @@ internal class ClusterClient : IInternalClusterClient
         private readonly IRuntimeClient runtimeClient;
         private readonly ClusterClientLifecycle clusterClientLifecycle;
         private readonly AsyncLock initLock = new AsyncLock();
+        private readonly ClientApplicationLifetime applicationLifetime;
```

Comment:
Why use concrete ClientApplicationLifetime  class when only ApplicationLifetime is necessary?

Reply:
Just to make it very clear that this will only be set when the host is the client